### ISSUE TITLE
Question: How to embed an interactive QtWidget into VSCode Plots pane?

### ISSUE THREAD

**Issue by ejmeitz** [julia-vscode/julia-vscode#3377]
I'm wrapping a Python library in Julia that is used to visualize atomic systems and I'd like to embed an interactive preview of the system in VsCode. Currently, the QtWidget just launches as a separate window. 

I can show PNG of the rendered scene in VSCode by overloading the Base.show with image/png as the mime type, however, I was wondering if it is possible to do something similar with my Qt widget? Matplotlib plots can do this so its definitely possible just maybe not from within a Julia package. I can also get this interactive preview as a jupyter widget if there's some way to use that outside of an IJulia notebook.

Thanks

**Comment by davidanthoff** [comment]
In general you are limited to stuff that works in a browser, but if QtWidget has support for that it might work. We have a custom MIME type `juliavscode/html` and if you add support for that via `show` you can send arbitrary HTML into the plot view in VS Code.

**Comment by ejmeitz** [comment]
I'm fairly certain I could embed the Qt object in HTML.

Are you aware of how libraries like PlotlyJS.jl etc. get interactive windows to appear in VSCode instead of just a static image? I've combed through the source code there and the MIME types look completely normal. As far as I know most plot libraries use some kind of GUI backend for this, but how they render that out is beyond me.

**Comment by pfitzseb** [comment]
> I'm fairly certain I could embed the Qt object in HTML.

If that works, then you can just send that HTML to VS Code with the `juliavscode/html` mime type. Plotly in particular probably uses `application/vnd.plotly.v1+json`, which we special case (same goes for Vega).

**Comment by ejmeitz** [comment]
Yeah I'll try that, I've only found [this](https://wiki.python.org/moin/PyQt/Embedding%20Widgets%20in%20Web%20Pages) for Qt embeddings and you basically make your own MIME type for that. Seems to be an uncommon operation though. I think WASM is the preferred entry point for qt on the web.

Either way Ill try it out, thanks!
